sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/ContainerAPITest.groovy

```diff
@@ -1423,4 +1427,170 @@ class ContainerAPITest extends APISpec {
         then:
         thrown(IllegalArgumentException)
     }
+
+    def "Restore Container"() {
+        given:
+        def cc1 = primaryBlobServiceClient.getBlobContainerClient(generateContainerName())
+        cc1.create()
+        def blobName = generateBlobName()
+        cc1.getBlobClient(blobName).upload(defaultInputStream.get(), 7)
+        cc1.delete()
+        def blobContainerItem = primaryBlobServiceClient.listBlobContainers(
```

Comment:
Why do you need to do a listing here to get the name of the blobContainerItem? Seems like you could just access the container name from the client directly

Reply:
In order to restore container  I need two things - name and version. There doesn't seem to be different way to get version of deleted container other than listing.

Reply:
```  
client.listBlobContainers(listBlobContainersOptions, null).forEach(
    deletedContainer -> {
        BlobContainerClient blobContainerClient = client.undeleteBlobContainer(
            deletedContainer.getName(), deletedContainer.getVersion());
    }
);
```

Reply:
That's right. I forgot about the version